src/boot/shim.c

```diff
@@ -30,15 +30,44 @@ struct ShimLock {
         EFI_STATUS __sysv_abi__ (*read_header) (void *data, uint32_t datasize, void *context);
 };
 
+typedef struct {
+        EFI_STATUS (EFIAPI *LoadImage)(
+                bool BootPolicy,
+                EFI_HANDLE ParentImageHandle,
+                EFI_DEVICE_PATH *DevicePath,
+                void *SourceBuffer,
+                size_t SourceSize,
+                EFI_HANDLE *ImageHandle);
+        EFI_STATUS (EFIAPI *StartImage)(
+                EFI_HANDLE ImageHandle,
+                size_t *ExitDataSize,
+                char16_t **ExitData);
+        EFI_STATUS (EFIAPI *Exit)(
+                EFI_HANDLE ImageHandle,
+                EFI_STATUS ExitStatus,
+                size_t ExitDataSize,
+                char16_t *ExitData);
+        EFI_STATUS (EFIAPI *UnloadImage)(EFI_HANDLE ImageHandle);
+} ShimImageLoader;
```

Comment:
how come this doesn't require sysv_abi like the other shim protocol?

Reply:
I used that at first, but it didn't work for some reason on EDK2, it crashed. This version was copied straight from the EFI_BOOT_SERVICES definition in `src/boot/efi.h` as it's exactly the same signatures. But I've removed it as requested now.